apis/r/R/utils.R

```diff
@@ -1,3 +1,62 @@
+#' Check if object is empty
+#' @noRd
+is_empty <- function(x) {
+  switch(class(x)[1],
+    "data.frame" = nrow(x) == 0,
+    length(x) == 0
+  )
+}
+
+#' Check if a vector is named
+#' @noRd
+is_named <- function(x) {
+  !is.null(names(x))
+}
+
+is_named2 <- function(x) {
+  return(!is.null(x = names(x = x)) && all(nzchar(x = names(x = x))))
```

Comment:
Why a return here and not in the other ones?  Can is_named and is_named2 be collapsed or do we need two?

Reply:
`return` was a stylistic choice that I wrote out of habit. I did not write `is_named` and can remove the `return` (and named parameters)

`is_named2` is more ~complete~ strict than `is_named`; I don't know who wrote `is_named` and what their requirements were, whereas I desired the strictness that comes from also checking with `nzchar()`

```R
is_named(c("" = 1)) # TRUE
is_named2(c("" = 1)) # FALSE
```

Reply:
Hm. Maybe `git blame` can help, and keeping it to one would be nice.  Else maybe use a more evocative name that simply suffixing with '2'?

Reply:
Totally agree, @eddelbuettel.

Reply:
I've merged `is_named` and `is_named2` by adding a new parameter to enable the checking that `is_named2` provides. By default, it follows `is_named` behavior (although running tests using `is_named2` behavior still passed)